### PR TITLE
Precompute values for brightness, contrast, gamma, and inversion

### PR DESCRIPTION
**Pull request by jetic83**
to speed up filters

Precompution happens at creation of filters.

### CODE REVIEW

File: openseadragon-filtering.js [hunk 1]
```diff
@@ -273,14 +273,18 @@
                 throw new Error(
                     'Brightness adjustment must be between -255 and 255.');
             }
+            var precomputed_BRIGHTNESS = []
```
Comment (avandecreme) [inline openseadragon-filtering.js]: Could you name the variable `precomputedBrightness` instead? (And change for the other filters as well).

**Comment by avandecreme** [comment]
Good idea.

Is it noticeable for Brigthness, Contrast and Invert filters as well?

**Comment by jetic83** [comment]
It has of course only a small effect for brightness, contrast and invert alone, but the effect becomes more noticable when you use multiple of those filters together.

**Comment by avandecreme** [comment]
Alright, thanks for doing this.

**Pull request merged**